libcxx/include/__random/uniform_real_distribution.h

```diff
@@ -27,6 +28,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 template<class _RealType = double>
 class _LIBCPP_TEMPLATE_VIS uniform_real_distribution
 {
+     static_assert(std::is_floating_point<_RealType>::value, "result_type must be a floating point type");
```

Comment:
This needs a test. You can take a look at `libcxx/test/libcxx/input.output/iostream.format/input.streams/traits_mismatch.verify.cpp` for an example of how to check static assertions.